standards/ICRC-3/ICRC-3.did

```diff
@@ -0,0 +1,56 @@
+type Value = variant {
+    Blob : blob;
+    Text : text;
+    Nat : nat; // do we need this or can we just use Int?
+    Int : int;
+    Array : vec Value;
+    Map : vec record { text; Value };
+};
+
+type GetArchivesArgs = record {
+    // The last archive seen by the client.
+    // The Ledger will return archives coming
+    // after this one if set, otherwise it
+    // will return the first archives.
+    from : opt principal;
+};
+
+type GetArchivesResult = vec {
+    // The id of the archive
+    canister_id : principal;
+
+    // The first block in the archive
+    start : nat;
+
+    // The last block in the archive
+    end : nat;
+}
+
+type GetBlocksArgs = vec record { start : nat; length : nat };
```

Comment:
@MarioDfinity Do you think it is a good idea to paginate the archives over here, like in the icrc3_get_archives function? I can picture a common scenario where a client gets the latest block-id with get_tip_certificate and then queries for get_blocks with arg start: 0, len: tip_certificate.last_block_index which returns all the current blocks in the main-ledger and the archived_blocks with every archive canister and their ranges. Since you put this pagination on the icrc3_get_archives method, I'm thinking it might be a good idea to paginate the archives here too. What do you think?   

Feel free to change this suggestion however you feel is best. 

```suggestion
type GetBlocksArgs = vec record { start : nat; length : nat; from_archive: opt principal };
```

Reply:
The get_blocks response already has a chunking mechanism for both, the blocks that the ledger itself contains and for the archives. If the archives don't fit into one response, then the ledger can return a callback to itself with the next range and then return the archives for that next range. So the above comment is not an issue.